programs/ssl/ssl_client2.c

```diff
@@ -594,7 +597,7 @@ int main( int argc, char *argv[] )
     memset( (void * ) alpn_list, 0, sizeof( alpn_list ) );
 #endif
 
-    if( argc == 0 )
+    if( argc <= 1 )
```

Comment:
Do we require that the output file be specified? If not, why do we need to change the minimum required number of arguments?

Reply:
The original value of argc is always greater than 0, because the first argument passed to argv[] is always the path to the program itself.
I don't understand why you determine if it is equal to 0, which makes no sense.

Reply:
OK, this is another fix that should go in its own commit. So far this PR looks like it fixes 2 issues (header, argc) and also adds a new feature; I'd expect 3 commits.

Reply:
OK. As your wish. code changed.